versions/draft-3/SPEC.md

```diff
@@ -240,30 +240,37 @@ $float = (([0-9]+)?\.([0-9]+)|[0-9]+\.|[0-9]+)([eE][-+]?[0-9]+)?
 
 ### Types
 
-All inputs and outputs must be typed.
+All inputs and outputs must be typed. The following primitive types exist in WDL:
 
+```wdl
+Int i = 0                  # An integer value
+Float f = 27.3             # A floating point number
+Boolean b = true           # A boolean true/false
+String s = "hello, world"  # A string value
+File f = "/path/to/file"   # A file
 ```
-$type = ($primitive_type | $array_type | $map_type | $object_type) $type_postfix_quantifier?
-$primitive_type = ('Boolean' | 'Int' | 'Float' | 'File' | 'String')
-$array_type = 'Array' '[' ($primitive_type | $object_type | $array_type) ']'
-$object_type = 'Object'
-$map_type = 'Map' '[' $primitive_type ',' ($primitive_type | $array_type | $map_type | $object_type) ']'
-$type_postfix_quantifier = '?' | '+'
+
+In addition, the following compound types can be constructed, parameterized by other types (in the examples below `X` and `Y` represent any valid type, even nested compound types):
+```wdl
+Array[X] xs = [x1, x2, x3]                    # An array of Xs
+Map[X,Y] x_to_y = { x1: y1, x2: y2, x3: y3 }  # A map from Xs to Ys
```

Comment:
In this example here, `X` Does not represent any Valid Type but a `primitive_type` as defined by hermes. So the constraint on X is that it must be: `Boolean`, `Int`, `Float`, `File`, or `String`. But the statement on line 253 would lead me to think that `X` can be a compound type as well, whish is false.